src/services/api/tests/api.unit.test.js

```diff
@@ -0,0 +1,258 @@
+import fetchMock from 'fetch-mock';
+import { Response } from 'node-fetch';
+import * as api from '../apiService';
+import config from '~/config'; // TODO: consider mocking this
+import {
+  ApiError,
+  GENERIC_ERROR_MESSAGE,
+  NetworkError,
+  TimeoutError
+} from '~/services/api/httpErrors';
+import store from '~/store';
+import { combineURLs } from '~/services/api/utils';
+
+// dedupe values used in multiple places within the test file
+const globals = {
+  randomRoute: 'fakeEndpoint',
+  randomJson: { hello: 'world' },
+  compileAbsoluteRoute: (relativeRoute) =>
+    combineURLs(config.apiUrl, relativeRoute)
+};
+
+describe('API module', () => {
+  afterEach(() => {
+    fetchMock.restore();
+    jest.restoreAllMocks();
+  });
+
+  describe('Generic request handler', () => {
+    it('prefixes urls with the api base url defined in the app config', async () => {
+      fetchMock.mock(`end:${globals.randomRoute}`, {});
```

Comment:
First of all, `end:`is a matcher offered by [fetch-mock](http://www.wheresrhys.co.uk/fetch-mock/).

We want to match all requests ending with the relative route used in the test since the api will prefix it with a baseUrl [here](https://github.com/FixMyBerlin/fixmy.frontend/pull/370/files#diff-7eedf2f3e1f7c249d3f4cdccfa29137fR18)

Since that is not clear, I [added a comment ](https://github.com/FixMyBerlin/fixmy.frontend/pull/370/commits/7050ffd9e615340044ce1ab4d5d79bf681d6029d).